joomla/joomla-cms
# Update README.md for alternate process using Docker for git and build

This is NOT urgent for 4.1.0 release.

And it looks like it requires some instructions adaptation for Windows.

Pull Request for Issue # none

### Summary of Changes

Adds alternate instructions using Docker Engine instead of installing git, php-cli, composer and Node.js+NPM.

I had issues installing Node.js into a test-environment to run very latest code for 4.1 tests, so I used Docker to run npm it quickly without installing it. Then I thought that an alternate run from sources using just Docker Engine instead of installing git, php-cli, composer and Node.js would be quite cool and modern-looking. While getting the exact docker command-lines can be and was a bit tricky, once they are known, this should run on any operating system.

These alternate instructions to run from sources could lower a bit one of the barriers to entry to contributors.

### Testing Instructions

1. Proofread English of changes (I'm not English-native).
2. Try it out:

- Install Docker Engine (or Docker Desktop) using the method of the URL provided. 
- Try out the alternate instructions, then 
- `ls -al` inside joomla-cms to check that the Joomla folders are there
- `ls -al libraries/vendor/` should show around 50 folders with your own ownership/permissions
- `ls -al media/vendor/debugbar/` should show around 13 files and 2 folders with your own ownership/permissions
- move joomla-cms folder inside your website directory and change folder permissions so that it can run in your enviroment
- try running joomla from that folder, it should provide the installation screen, and run fine.
- Provide feedback of the operating systems you tried and your test results.

Ideally, we need tests in MacOS and Windows too. I tested it successfully on Linux (latest Ubuntu LTS (20.04))

### Actual result BEFORE applying this Pull Request

No documented way to install from sources without installing php-cli, composer and Node.js with npm software natively and permanently into your system.

### Expected result AFTER applying this Pull Request

1. Documented way to install from sources without installing any software (appart from Docker Engine) in the README.md.

2. The documented Docker-assisted installation from source works.

### Documentation Changes Required

None needed, This, README.md, *is* just the documentation.
